src/uiSelectDirective.js

```diff
@@ -206,11 +206,15 @@ uis.directive('uiSelect',
           $select.clickTriggeredSelect = false;
         }
 
-        // See Click everywhere but here event http://stackoverflow.com/questions/12931369
-        $document.on('click', onDocumentClick);
+        // See Click everywhere but here. Similar approach to http://stackoverflow.com/questions/12931369
+        // but using the capture phase instead of bubble phase of the event propagation.
+        //
+        // Using the capture phase avoids problems that araise when event.stopPropatagion()
+        // is called before the event reaches the `document`.
+        $window.document.addEventListener('click', onDocumentClick, true);
```

Comment:
In [uiSelectchoicesDirective.js#L51](https://github.com/nzzdev/ui-select/blob/6f19d5fb738c75c960a6cfc4eda0bfa6a0769350/src/uiSelectChoicesDirective.js#L51) ```$window.document``` is used, so i wanted to be consistent.

Also it is not possible to add a native ```addEventListener``` to the angular $document, which is a jQuery [```window.document```](https://docs.angularjs.org/api/ng/service/$document) wrapper. 
Attaching native eventListener to jQuery-objects is not possible, which is confirmed [for example here](https://stackoverflow.com/questions/21680825/event-capturing-event-bubbling-and-jquery-on).

jQuery's aim is browser compability and capture phase is not supported by older IE's. Because Angular only [supports IE9+](https://angular.io/docs/ts/latest/guide/browser-support.html) this should not be an issue.